src/main/java/com/hazelcast/kubernetes/KubernetesApiEndpointResolver.java

```diff
@@ -56,49 +53,59 @@
     @Override
     List<DiscoveryNode> resolve() {
         if (serviceName != null && !serviceName.isEmpty()) {
-            return getSimpleDiscoveryNodes(client.endpointsByName(namespace, serviceName));
+            return getSimpleDiscoveryNodes(client.endpointsByName(serviceName));
         } else if (serviceLabel != null && !serviceLabel.isEmpty()) {
-            return getSimpleDiscoveryNodes(client.endpointsByLabel(namespace, serviceLabel, serviceLabelValue));
+            return getSimpleDiscoveryNodes(client.endpointsByLabel(serviceLabel, serviceLabelValue));
         }
-        return getSimpleDiscoveryNodes(client.endpoints(namespace));
+        return getSimpleDiscoveryNodes(client.endpoints());
     }
 
-    private List<DiscoveryNode> getSimpleDiscoveryNodes(Endpoints endpoints) {
+    private List<DiscoveryNode> getSimpleDiscoveryNodes(List<Endpoint> endpoints) {
         List<DiscoveryNode> discoveredNodes = new ArrayList<DiscoveryNode>();
-        resolveNotReadyAddresses(discoveredNodes, endpoints.getNotReadyAddresses());
```

Comment:
I see `resolveNotReadyAddresses` is not used anymore so there might be more cleanup. You can even remove `resolveAdresses` and move up the for loop inside `getSimpleDiscoveryNodes`

Reply:
Good point. Changed.